packages/form/src/IconField.tsx

```diff
@@ -10,23 +12,49 @@ type IconFieldProps = ComponentProps<'div'> & {
   icon: ReactElement<IconProps>;
   children: JSX.Element | JSX.Element[];
   'data-test-id'?: string;
+  tooltip?: string | JSX.Element;
+  renderIconLast?: boolean;
+  ariaLabel?: string;
 };
 
 const IconField = ({
   icon,
   children,
   className,
   'data-test-id': testId = 'icon-field',
+  tooltip,
+  renderIconLast = false,
+  ariaLabel = 'More info',
   ...rest
 }: IconFieldProps) => {
-  const renderIcon = cloneElement(icon, { size: 'small', className: styles.iconFieldIcon });
+  const iconElement = cloneElement(icon, {
+    size: 'small',
+    className: cx(styles.iconFieldIcon, styles.iconFieldIconFill),
+  });
 
-  const classes = cx(styles.iconField, className);
+  const classes = cx(styles.iconField, renderIconLast ? 'IconAfter' : 'IconBefore', className);
+
+  const renderIcon = tooltip ? (
+    <Tooltip content={tooltip} targetClassName={styles.iconFieldButton}>
+      <IconButton
```

Comment:
Did you mean to make this into a button? Will users interact with it? (It looks like you can't pass `onClick` to this specific component for example.)

Reply:
Yeah tooltips must have an interactable element as the target/trigger. `onClick` isn't needed as `Tooltip` already adds listeners for hover/focus to show the content.